File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l = 0;
        int r = 0;
        int sum = 0;
        int res = Integer.MAX_VALUE;

        for(int i=0;i<nums.length;i++){

            sum+= nums[i];

            while(sum>=target) {

                res = Math.min(res,r-l+1);
                sum -= nums[l];
                l++;
            }

            r++;
        }
        if(res==Integer.MAX_VALUE) return 0;
        return res;
        
    }
}

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(tar, nums) {

    let l = r = 0;
    let sum = 0;
    let res = Infinity;

    for(let i=0;i<nums.length;i++){
        sum+=nums[i];

        while(sum>=tar){

            res = Math.min(res,r-l+1)
            sum = sum - nums[l];
            l++
        }
        r++;
    } 
      if(res==Infinity) return 0;
    return res;
    
};